cmd/call-endpoint/main.go

```diff
@@ -0,0 +1,172 @@
+
+		// Return to prevent further execution after log.Fatal().
+		return
+	}
+
+	var endpointDescription *core.EndpointDescription
+
+	apiDescription := api.Describe(*api.GetRoutes())
+	for endpoint, requestResponse := range apiDescription.Endpoints {
+		if endpoint == args.Endpoint {
+			endpointDescription = &requestResponse
+			break
+		}
+	}
+
+	if endpointDescription == nil {
+		log.Fatal("Failed to find the endpoint.", log.NewAttr("endpoint", args.Endpoint))
+
+		// Return to prevent further execution after log.Fatal().
+		return
+	}
+
+	request := map[string]any{}
+	for _, arg := range args.Parameters {
+		// Split the parameter into it's key and value.
+		parts := strings.SplitN(arg, ":", 2)
+		if len(parts) != 2 {
+			log.Fatal("Invalid parameter format: missing a colon. Expected format is 'key:value', e.g., 'id:123'.", log.NewAttr("parameter", parts))
+
+			// Return to prevent further execution after log.Fatal().
+			return
+		}
+
+		request[parts[0]] = parts[1]
+	}
+
+	var printFunc cmd.CustomResponseFormatter
+	if args.Table {
+		printFunc = printCMDResponseTable
+	}
+
+	cmd.MustHandleCMDRequestAndExitFull(args.Endpoint, request, nil, args.CommonOptions, printFunc)
+}
+
+func generateHelpDescription() string {
+	baseDescription := "Execute an API request to the specified endpoint.\n\n"
+
+	var endpointList strings.Builder
+	endpointList.WriteString("List of endpoints:\n")
+
+	apiDescription := api.Describe(*api.GetRoutes())
+	for endpoint := range apiDescription.Endpoints {
+		endpointList.WriteString(fmt.Sprintf("  - %s\n", endpoint))
+	}
+
+	return baseDescription + endpointList.String()
+}
+
+func printCMDResponseTable(response core.APIResponse) string {
+	responseContent, ok := response.Content.(map[string]any)
+	if !ok {
+		return ""
+	}
+
+	// Don't try to format a response that has multiple keys.
+	if len(responseContent) != 1 {
+		return ""
+	}
+
+	users, ok := responseContent[USERS_KEY].([]any)
+	if !ok {
+		return ""
+	}
+
+	firstUser, ok := users[0].(map[string]any)
+	if !ok {
+		return ""
+	}
+
+	var headers []string
+	for key := range firstUser {
+		if key == COURSES_KEY {
+			continue
+		}
+
+		headers = append(headers, key)
+	}
+
+	sort.Strings(headers)
+
+	_, exists := firstUser[COURSES_KEY]
+	if exists {
+		// Add courses to the end of the slice for better readability in the output.
+		headers = append(headers, COURSES_KEY)
+	}
+
+	var usersTable strings.Builder
+	usersTable.WriteString(strings.Join(headers, "\t"))
+
+	headerKeys := strings.Split(usersTable.String(), "\t")
+
+	usersTable.WriteString("\n")
+
+	for i, user := range users {
+		userMap, ok := user.(map[string]any)
+		if !ok {
+			return ""
+		}
+
+		var row []string
+		for _, key := range headerKeys {
+			switch value := userMap[key].(type) {
+			case string:
+				row = append(row, value)
+			case int:
+				row = append(row, strconv.Itoa(value))
+			case bool:
+				row = append(row, strconv.FormatBool(value))
```

Comment:
I'm not a fan of these cases.
What about float?
Why do the non-JSON cases have to be treated differently?